maliput_ros_translation/src/maliput_ros_translation/convert.cc

```diff
@@ -253,4 +255,90 @@ maliput::api::RoadPosition FromRosMessage(const maliput::api::RoadGeometry* road
+  msg.z = rotation.quat().z();
+  msg.w = rotation.quat().w();
+  return msg;
+}
+
+maliput_ros_interfaces::msg::IsoLaneVelocity ToRosMessage(const maliput::api::IsoLaneVelocity& velocity) {
+  maliput_ros_interfaces::msg::IsoLaneVelocity msg;
+  msg.sigma_v = velocity.sigma_v;
+  msg.rho_v = velocity.rho_v;
+  msg.eta_v = velocity.eta_v;
+  return msg;
+}
+
+maliput::api::IsoLaneVelocity FromRosMessage(const maliput_ros_interfaces::msg::IsoLaneVelocity& velocity) {
+  return maliput::api::IsoLaneVelocity{velocity.sigma_v, velocity.rho_v, velocity.eta_v};
+}
+
+maliput_ros_interfaces::msg::HBounds ToRosMessage(const maliput::api::HBounds& h_bounds) {
+  maliput_ros_interfaces::msg::HBounds msg;
+  msg.min = h_bounds.min();
+  msg.max = h_bounds.max();
+  return msg;
+}
+
+maliput::api::HBounds FromRosMessage(const maliput_ros_interfaces::msg::HBounds& h_bounds) {
+  return maliput::api::HBounds(h_bounds.min, h_bounds.max);
+}
+
+maliput_ros_interfaces::msg::RBounds ToRosMessage(const maliput::api::RBounds& r_bounds) {
+  maliput_ros_interfaces::msg::RBounds msg;
+  msg.min = r_bounds.min();
+  msg.max = r_bounds.max();
+  return msg;
+}
+
+maliput::api::RBounds FromRosMessage(const maliput_ros_interfaces::msg::RBounds& r_bounds) {
+  return maliput::api::RBounds(r_bounds.min, r_bounds.max);
+}
+
+maliput_ros_interfaces::msg::SRange ToRosMessage(const maliput::api::SRange& s_range) {
+  maliput_ros_interfaces::msg::SRange msg;
+  msg.s0 = s_range.s0();
+  msg.s1 = s_range.s1();
+  msg.size = s_range.size();
+  msg.with_s = s_range.WithS();
+  return msg;
+}
+
+maliput::api::SRange FromRosMessage(const maliput_ros_interfaces::msg::SRange& s_range) {
+  return maliput::api::SRange(s_range.s0, s_range.s1);
+}
+
+maliput_ros_interfaces::msg::LaneSRange ToRosMessage(const maliput::api::LaneSRange& lane_s_range) {
+  maliput_ros_interfaces::msg::LaneSRange msg;
+  msg.lane_id = ToRosMessage(lane_s_range.lane_id());
+  msg.s_range = ToRosMessage(lane_s_range.s_range());
+  return msg;
+}
+
+maliput::api::LaneSRange FromRosMessage(const maliput_ros_interfaces::msg::LaneSRange& lane_s_range) {
+  return maliput::api::LaneSRange(FromRosMessage(lane_s_range.lane_id), FromRosMessage(lane_s_range.s_range));
+}
+
+maliput_ros_interfaces::msg::LaneSRoute ToRosMessage(const maliput::api::LaneSRoute& lane_s_route) {
+  maliput_ros_interfaces::msg::LaneSRoute msg;
+  msg.ranges.resize(lane_s_route.ranges().size());
+  std::transform(lane_s_route.ranges().cbegin(), lane_s_route.ranges().cend(), msg.ranges.begin(),
+                 [](const maliput::api::LaneSRange& lane_s_range) { return ToRosMessage(lane_s_range); });
+  msg.length = lane_s_route.length();
+  return msg;
+}
+
+maliput::api::LaneSRoute FromRosMessage(const maliput_ros_interfaces::msg::LaneSRoute& lane_s_route) {
+  std::vector<maliput::api::LaneSRange> ranges;
+  // Not using std::transform() because we ranges should be resized and it fails to compile due to LaneId
+  // LaneSRange.
```

Comment:
```suggestion
  // Not using std::transform() because the ranges should be resized and it fails to compile due
  // to LaneId in LaneSRange.
```

Reply:
Is it because something discarding qualifiers?

Reply:
We need to call std::vector<T>::resize() which requires T to be DefaultInsertable. It means to be DefaultConstructible and TypeSpecificSpecifier<T> is not. It requires a non-empty string.